addons/event_sale/tests/test_event_sale.py

```diff
@@ -376,7 +384,200 @@ def test_ticket_price_with_pricelist_and_tax(self):
+        company_currency = self.env.company.currency_id
+        other_currency = eur_currency if company_currency != eur_currency else usd_currency
+        precision_delta = other_currency.rounding
+        price_list = self.env['product.pricelist'].create({
+            'name': 'Sale pricelist',
+            'discount_policy': 'with_discount',
+        })
+        tax = self.env['account.tax'].create({
+            'name': "Tax 10",
+            'amount': 10,
+        })
+        event = self.env['event.event'].create({
+            'name': 'New Event',
+            'date_begin': '2020-02-02',
+            'date_end': '2020-04-04',
+        })
+        now = datetime.datetime.now()
+
+        def create_sale_order(is_product_templ,
+                              product_lst_price, ticket_price, qty,
+                              rule_type, rule_amount, min_qty,
+                              rule_based_on=None, product_standard_price=10,
+                              price_list_currency=None):
+            """Create a sale order and related instances based on the provided configuration.
+
+            :param bool is_product_templ: whether to use a product template or a product
+            :param float product_lst_price: price of the product linked to the event ticket
+            :param float ticket_price: price of the ticket
+            :param int qty: number of ticket ordered in the sale order line
+            :param str rule_type: price list item compute_price (formula, fixed or percentage)
+            :param float rule_amount: amount used for the rule_type (ex.: rule_type = fixed, it means the fixed price)
+            :param int min_qty: minimum quantity for the rule to be applied
+            :param str rule_based_on: by default on the ticket price (list_price), can be set to standard_price (cost)
+            :param float product_standard_price: product standard_price (product cost which is used as ticket cost)
+            :param res_currency price_list_currency: currency of the price list
+            :return: (sale_order, sale_order_line)
+            """
+            if price_list_currency:
+                price_list.write({'currency_id': price_list_currency.id})
+            if is_product_templ:
+                product_templ = self.env['product.template'].create({
+                    'name': 'event_product',
+                    'detailed_type': 'event',
+                    'standard_price': product_standard_price,
+                    'list_price': product_lst_price,
+                    'taxes_id': [Command.set([tax.id])],
+                })
+            else:
+                product = self.env['product.product'].create({
+                    'name': 'event_product',
+                    'detailed_type': 'event',
+                    'lst_price': product_lst_price,
+                    'standard_price': product_standard_price,
+                    'taxes_id': [Command.set([tax.id])],
+                })
```

Comment:
the created values are the same, you could define the values once, then use it to create a single 'product_or_template':
```python
            product_or_template_values = {
                'name': f"event_product{'_template' if is_product_template else ''}",
                'detailed_type': 'event',
                'list_price': product_lst_price,
                'standard_price': product_standard_price,
                'taxes_id': [Command.set([tax.id])],
            }
            if is_product_templ:
                product_or_template = self.env['product.template'].create(product_or_template_values)
            else:
                product_or_template = self.env['product.product'].create(product_or_template_values)
```
Then use product_or_template for the event ticket and others like
```python
self.env['event.event.ticket'].create({
    ...
    'product_id': product_or_template.product_variant_id.id
})
```

Reply:
Good idea, thanks.